mb-store/src/core.rs

```diff
@@ -296,15 +340,46 @@ impl MintbaseStore {
     }
 }
 
+/// Checks if `account_id` is allowed to transfer the token and returns the
+/// `authorized_id` to log. Explicitly, returns `None` if token is owned by
+/// `account_id`, returns `Some(account_id)` if `account_id` was approved
+/// with the correct `approval_id`, panics otherwise.
+fn assert_token_owned_or_approved(
+    token: &Token,
+    account_id: &AccountId,
+    approval_id: Option<u64>,
```

Comment:
When implementing a POC level integration (NoRamp), I did run into a situation where the approved approved system did not retain the approval id.  Not sure such a case will come again, but this check was too strict to handle that instance.